app/controllers/concerns/api/v1/doorkeeper_authentication.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    module DoorkeeperAuthentication
+      extend ActiveSupport::Concern
+      include API::V1::ErrorHandler
+
+      included do
+        before_action :doorkeeper_authorize!
+      end
+
+      private
+
+      def current_user
+        User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+      end
+
+      def doorkeeper_unauthorized_render_options(error: nil)
+        return unless error
+
+        {
+          json: {
+            errors: build_errors(detail: error.description, source: error.state, code: error.name)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/ddfec4932dbb11ae78f67f52b257b3efda688cc1/app/controllers/concerns/api/v1/doorkeeper_authentication.rb#L24"></span>Refers to ‘error’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>